### PR TITLE
MRG Butterfly epoch

### PR DESCRIPTION
**Pull request by jaeilepp**
Butterfly plot can be toggled with a 'b' key.
closes https://github.com/mne-tools/mne-python/issues/2205

### CODE REVIEW

**Comment by mainakjas** [comment]
@jaeilepp can we have a help pop-up when you press `?` a la gmail? I think @choldgraf @teonlamont and myself all agree that this would be useful.

**Comment by agramfort** [comment]
can you share screenshots to it's easy to have a feeling of what you're working on?

**Comment by jaeilepp** [comment]
Currently working on y axis...
screenshot from 2015-06-16 15 20 13

**Comment by agramfort** [comment]
you're going to make 3 butterflies? one per channel type?

**Comment by jaeilepp** [comment]
There's a button for changing which channel type to view. Default view is the one with all channels.

**Comment by mainakjas** [comment]
hmm ... do we really need a button? can't we just have it all there as three subplots?

**Comment by jaeilepp** [comment]
> can't we just have it all there as three subplots?

Good point. What about the other types (EOG etc.). Should they be shown too?

**Comment by agramfort** [comment]
EOG should be displayed separately

**Comment by mainakjas** [comment]
I was thinking more along the lines of `evoked.plot()` which has `grad`, `mag` and `EEG` nothing else. If we start looking at EOG, ECG, we have tons of other channel types ...

**Comment by jaeilepp** [comment]
So what's the verdict? Three axes with mag, grad and eeg, fourth with eog or one for all existing types?

**Comment by mainakjas** [comment]
I'll let @agramfort decide

**Comment by dengemann** [comment]
> Three axes with mag, grad and eeg, fourth with eog or one for all existing types?

Yes, this should be it!

**Comment by mainakjas** [comment]
I don't quite like the toggle key to be 'b' because it doesn't make sense when you want to go back from the butterfly plot. How about something like `spacebar` (or any other better suggestions ...) ?

**Comment by choldgraf** [comment]
+1 to a pop up infobox with possible key presses. Personally I think 'b' would be fine if the option were given as "turn butterfly plot on/off". Then it's just a toggle and semantically makes sense to be 'b' for me.

**Comment by jaeilepp** [comment]
Ready for review, I guess. Unless we want the infobox for this PR.

**Comment by dengemann** [comment]
> Unless we want the infobox for this PR.

infobox later.

**Comment by dengemann** [comment]
@jaeilepp trying to run the example on your branch, it seems there are regressions, 

```
/Users/dengemann/github/mne-python/examples/plot_from_raw_to_epochs_to_evoked.py in <module>()
     51 
     52 # Plot epochs.
---> 53 epochs.plot(trellis=False, block=True, fig_title='Auditory left/right')
     54 
     55 # Look at channels that caused dropped events, showing that the subject's

TypeError: plot() got an unexpected keyword argument 'fig_title'
```

Also the figure looks weird now:
https://www.dropbox.com/s/r9sl6c655bn3osc/Screenshot%202015-06-17%2012.34.38.png?dl=0

**Comment by dengemann** [comment]
Butterfly view looks awesome!

I really much more liked the event ids on the top x axis, it was much cleaner.

**Comment by jaeilepp** [comment]
> I really much more liked the event ids on the top x axis, it was much cleaner.

Yeah, especially when there are a lot of epochs/view. I should probably change it back and deal with the layout problem of the title on another PR.

**Comment by teonlamont** [comment]
it'll be nice to have the infobox, I honestly have no clue what key does what other than the '-', '=', and '+'.

**Comment by mainakjas** [comment]
yeah +100 on infobox

**Comment by agramfort** [comment]
screenshot? :)

**Comment by jaeilepp** [comment]
screenshot from 2015-06-17 14 41 21

**Comment by mainakjas** [comment]
The channel bar shouldn't be there any more no?

**Comment by mainakjas** [comment]
how do I know which is grad, mag or EEG?

**Comment by mainakjas** [comment]
no units on y axis?

**Comment by mainakjas** [comment]
can we have it interactive so that you can click and know the channel name, just like in `evoked.plot()` ?

**Comment by teonlamont** [comment]
for the title, does that require a separate pr to fix? it's reverted to the clustering in the top. @dengemann my two concerns about the event id at the top is that it is clutters the title area at the moment, and it's not labeled. i know that @jaeilepp mentioned that to get the layout right with the title and the event ids at the top, he would need to rework the shared layout for both raw.plot and epochs.plot

**Comment by jaeilepp** [comment]
> can we have it interactive so that you can click and know the channel name, just like in evoked.plot() ?

What would be the button? Left click selects a bad epoch and right click draws a line.

**Comment by dengemann** [comment]
But we outvoted you so far. I'm not happy about the fact that you just undo what we discussed. Second,  having only the triggers is not ideal but a good compromise as it will be in most cases 2 digits and nothing else. The current proposal is barely parseable and cannot be considered an alternative.

**Comment by teonlamont** [comment]
looking at epochs.plot more and more, the two vertical lines (dashed epoch boundaries and event trigger) really makes the plot look jumbled. the alternating coloring make the boundaries more salient. what was the argument against it?

**Comment by teonlamont** [comment]
> But we outvoted you so far. I'm not happy about the fact that you just undo what we discussed. Second,  having only the triggers is not ideal but a good compromise as it will be in most cases 2 digits and nothing else. The current proposal is barely parseable and cannot be considered an alternative.

I didn't make any changes, all i did was raise my concerns about the title/event id. I didn't say we needed to go back to that one, I was merely pointing out that adding the event id back to the top does not solve the initial issue I mentioned about the crowding of the title and the event ids.

**Comment by dengemann** [comment]
Complementary color after inages, consistency with eeglab, simplicity, hours of viewing experience. You want the signals nothing else. And you need some visual clustering. A line is the least invasive you can have.

**Comment by dengemann** [comment]
We need a different solution, e.g. adding the title above (the title is IMO less important for interpreting the plot than are the event ids). For the labels I'm still not sure. Their length is user dependent. It will be a mess. Maybe an auxiliary figure? We don't really have room for the event id legend inside the plot. I fear what we have so far is already a good compromise.

**Comment by teonlamont** [comment]
> We need a different solution, e.g. adding the title above (the title is IMO less important for interpreting the plot than are the event ids). For the labels I'm still not sure. Their length is user dependent. It will be a mess. Maybe an auxiliary figure? We don't really have room for the event id legend inside the plot. I fear what we have so far is already a good compromise.

The title is there to be consistent with raw.plot. I totally agree that the event ids are useful, it would be cleaner to have some space between the title and the labels, that's it. I don't think we need to put the legend there, the numbers are enough, I think.

**Comment by jaeilepp** [comment]
The current layout configuration that is shared with raw.plot uses inches in determining the size for some reason. That is dependent on the screen size (I think), so it probably should be changed to use pixels instead. I suspect its going to be a messy project so I'll preferably leave it for another PR.

**Comment by teonlamont** [comment]
ok, that's fine with me. just pointing it out

**Comment by jaeilepp** [comment]
I added the possibility to add the label on butterfly with middle mouse click (or wheel click). I'm not sure if it's of much use, but we already have functions for both of the other mouse buttons.
screenshot from 2015-06-18 13 48 31

**Comment by agramfort** [comment]
can you also test with somato dataset and spm dataset?

**Comment by teonlamont** [comment]
is it possible to rotate the y-label -90 degrees and for the 0.00 to be an int 0?

**Comment by mainakjas** [comment]
On Thu, Jun 18, 2015 at 12:49 PM, jaeilepp <notifications@github.com> wrote:

> I added the possibility to add the label on butterfly with middle mouse
> click (or wheel click). I'm not sure if it's of much use,
>
I think it's pretty useful to find outlier channels. Is it possible to
change the color of the selected channel? ... and maybe make it's zorder to
-1. I am not sure what will look the best but you get the idea I hope. Any
chance you can use modifier keys to work? It's only a problem with macosx
backend? We decided to drop support for it anyway? So, maybe we can have
shift + left click for this? What do you think (let's discuss it first
before you implement it)?

**Comment by mainakjas** [comment]
The ylabel shouldn't say mag/grad etc. because that's not what the y-axis shows. It should be as titles/subtitles.

**Comment by mainakjas** [comment]
The colors aren't encoding anything anymore. Should we make them all black or do you want to keep the same colors as in the other plot?

**Comment by mainakjas** [comment]
why are the plots going above the vertical lines. Shouldn't it be the other way round?

**Comment by mainakjas** [comment]
For the units, can you use `uV` `fT` etc so you don't have to put exponents in the yticks. I think that's how it's done in `evoked.plot()`

**Comment by jaeilepp** [comment]
> Any chance you can use modifier keys to work? It's only a problem with macosx backend? We decided to drop support for it anyway?

I think all of the current features (apart from f11) work on osx too. I think we should stay away from modifier keys as long as we can.

**Comment by mainakjas** [comment]
hmm ... okay. Middle mouse key won't do it I think. I have no idea how to have it on a trackpad. Let's wait for other opinions

**Comment by teonlamont** [comment]
> We decided to drop support for it anyway?

just for fullscreen there's a problem

**Comment by Eric89GXL** [comment]
I think it's okay for the labels for the different channel types to be along the y axis if necessary. I think creating different subplots just to move those titles will be pain. Plus you can think of it as the different channel types actually spanning the y axis range, so it is somewhat justifiable to put the labels along the y axis :)

**Comment by mainakjas** [comment]
no need to create different subplots. Can't we use text annotations which look like titles?

**Comment by teonlamont** [comment]
also, it would be nice to be able to select bad channels in the epochs.plot viewer. Epochs are sometimes entry points for some data processing.

**Comment by Eric89GXL** [comment]
@mainakjas yes we could. We can try it, but I suspect in some plots the data will run into the text objects since they will be in the data area, so it might trade a controlled aesthetic problem into an uncontrolled / difficult one.

**Comment by jaeilepp** [comment]
I had some trouble with the units. Could someone check that they seem reasonable? I don't have any data sets for testing it myself at the moment.

**Comment by teonlamont** [comment]
since there is now the toggle to butterfly, we can now make it the mne_browse_epochs viewer the default, trellis=False

**Comment by teonlamont** [comment]
also, if there is only one channel type in the data, there should only be a single subplot on the axis

**Comment by teonlamont** [comment]
also, I can toggle from the channel view to the butterfly plot but if I zoom on the butterfly and toggle back, none of the channels are shown

**Comment by mainakjas** [comment]
I think the zoom should be preserved in the original view even if you scale up / down in butterfly mode

**Comment by mainakjas** [comment]
the zooming appears to be slow in butterfly mode. cc @teonlamont

**Comment by teonlamont** [comment]
the default number of epochs should be fewer than what we have now, 20 --> 5.

currently the butterfly plot is a bit slow, this is probably because of the number of channels that are being plotted. i don't have a suggestion on making it faster. @mainakjas mentioned maybe antialising?

**Comment by Eric89GXL** [comment]
Yes, disable aa, it will be faster

**Comment by dengemann** [comment]
> I think it's okay for the labels for the different channel types to be along the y axis if necessary. I think creating different subplots just to move those titles will be pain.

I think for now y-labels are ok if it helps having the text outsde of the plot.

> also, it would be nice to be able to select bad channels in the epochs.plot viewer. Epochs are sometimes entry points for some data processing.

will not be so easy as clicking is already for rejecting. wondering how fast it will be. would would be good is some annotation mode, where you can get epoch-wise and channel-wise user annotations.

**Comment by mainakjas** [comment]
It seems to me that the annotations themselves are not slowing things, it's the amount of data being plotted that makes everything slow

**Comment by teonlamont** [comment]
bad channels are being shown in butterfly mode. there should be an exclude argument like in Evoked.plot where bads are not displayed is `exclude='bads'` and are marked as red if `exclude=[]`

**Comment by dengemann** [comment]
Ahh ok, got it. I would vote for bad channels in the browser mode. Not sure
you will recognize single channels easily in the butterfly view on single
trials.

2015-06-18 21:20 GMT+02:00 Teon L Brooks <notifications@github.com>:

**Comment by dengemann** [comment]
But we should try it.

2015-06-18 21:23 GMT+02:00 Denis-Alexander Engemann <
denis.engemann@gmail.com>:

**Comment by agramfort** [comment]
works pretty well. Just one usability comment. When switching to butterfly the scaling seems too small so I increase it but I switch back to default view scale is too big. Can we have a highler scaling by default in the butterfly mode?

**Comment by jaeilepp** [comment]
Antialiasing is disabled. Butterfly plot is slower because of the amount of data being plotted. I don't think there's much I can do about it without slowing down the default view. I made it so that most of the heavy stuff is done when changing the view ('b' is clicked).

**Comment by agramfort** [comment]
works nicely for me !

+1 for merge if your done.

**Comment by agramfort** [comment]
nice work @jaeilepp !

File: examples/plot_from_raw_to_epochs_to_evoked.py [hunk 1]
```diff
@@ -50,7 +50,7 @@
                     preload=True)
 
 # Plot epochs.
-epochs.plot(trellis=False, block=True, fig_title='Auditory left/right')
+epochs.plot(trellis=False, block=True, title='Auditory left/right')
```
Comment (mainakjas) [inline examples/plot_from_raw_to_epochs_to_evoked.py]: don't we need to get rid of this argument now?

File: mne/viz/tests/test_epochs.py [hunk 2]
```diff
@@ -104,7 +104,7 @@ def test_plot_epochs():
     """Test epoch plotting"""
     import matplotlib.pyplot as plt
     epochs = _get_epochs()
-    epochs.plot(scalings=None, fig_title='Epochs', trellis=False)
+    epochs.plot(scalings=None, title='Epochs', trellis=False)
```
Comment (mainakjas) [inline mne/viz/tests/test_epochs.py]: no test to update?

**Comment by mainakjas** [comment]
after I close the plot, it doesn't return control to the command line for me

**Comment by mainakjas** [comment]
very minor nitpick, the order `mag`, `grad` then `EEG` is different from what we do in `evoked.plot()`. But I guess this is what we do in `raw.plot()` too, so it should be good ...

File: examples/plot_from_raw_to_epochs_to_evoked.py (hunk 1 above)
Comment (mainakjas) [inline examples/plot_from_raw_to_epochs_to_evoked.py]: ok I noticed a deprecation warning, so we're fine here I think

**Comment by mainakjas** [comment]
Add the tests @jaeilepp and you have my +1 for merge

**Comment by mainakjas** [comment]
... and yeah, nice work!

**Comment by jaeilepp** [comment]
Okay, ready.

**Comment by agramfort** [comment]
works great for me.

+1 for merge.

**Comment by teonlamont** [comment]
do we still need to show the other channel types if only one channel type is present in the epochs?

**Comment by agramfort** [comment]
indeed we should not

**Comment by teonlamont** [comment]
I think @Eric89GXL mentioned the ESC key. I don't know if it was for this pr or not, but it doesn't work to close the plot. I'm using the 'qt4' backend. if it's for #2214 then don't worry about it for now

**Comment by agramfort** [comment]
@jaeilepp did you try the code with the other datasets we have?
somato, spm CTF and just EEG?

**Comment by jaeilepp** [comment]
I tried with somato data set. Seems to work.

**Comment by agramfort** [comment]
and CTF?

**Comment by dengemann** [comment]
CTF: https://www.dropbox.com/s/4ye2lvuwskdwp3m/Screenshot%202015-06-22%2013.13.39.png?dl=0

I think we should not pick ref channels, also weird that the other channels don't contain data.

BTW @jaeilepp any reaosns why block=True in the basic example. I kind of overlooked it, this should not be.

**Comment by dengemann** [comment]
Note. I think we should only display the channels which are available. We should also display the ECG if available, or at least allow for displaying it.

**Comment by agramfort** [comment]
+100

**Comment by jaeilepp** [comment]
Are there sample data sets with ecg channels available?

**Comment by agramfort** [comment]
there is no ECG in somato?

**Comment by jaeilepp** [comment]
> there is no ECG in somato?

Nope.

**Comment by agramfort** [comment]
you could artificially create one just for debugging...

**Comment by dengemann** [comment]
@jaeilepp take a look at raw.set_channel_types -- pick a channel randomly and overwrite it with the mean across magnetometers which gives a nice ECG.

**Comment by jaeilepp** [comment]
I noticed that the ecg channels are colored red which is the same color we use for bad epochs, so there's no way of seeing bad epochs if you use only ecg channels (except the scroll bar). Also, I used uV as a unit for ecg. Is this okay?

**Comment by agramfort** [comment]
yes ECG is in uV

pick a different color then for ECG. purple?

**Comment by jaeilepp** [comment]
It's using the default channel colors from defaults module. If I change it, it would be inconsistent with other plots.

**Comment by jaeilepp** [comment]
Would it be okay to change the color in defaults module?

**Comment by agramfort** [comment]
yeah for ECG it should not be a problem.

**Comment by agramfort** [comment]
wait one more thing. You could also make the bad in light gray? and
keep the ECG red? can you try and share screenshots?

**Comment by jaeilepp** [comment]
I changed it to magenta as purple seemed too dark to my eye. Light gray is used for bad channels, but I don't know if it's a problem if they intermingle. 

screenshot from 2015-06-23 10 54 43

For some reason my print screen crops the screen...

**Comment by agramfort** [comment]
ok ok. Maybe you can keep the red for both as it means anything that is red
will be ignored. makes sense?

**Comment by jaeilepp** [comment]
You mean like this?
screenshot from 2015-06-23 11 29 26

**Comment by agramfort** [comment]
hum... why not making bads light gray as in browse_raw?

**Comment by mainakjas** [comment]
also can the channel names of the bads also be made light gray?

**Comment by mainakjas** [comment]
and can you make the channel on top of the dashed vertical line ?

**Comment by jaeilepp** [comment]
> also can the channel names of the bads also be made light gray?

They can be made light gray, but that would mean extra iterations over the labels and colors every time view is changed, because of the way the colors are defined at the moment. Not worth the effort/decrease in performance in my opinion.

**Comment by jaeilepp** [comment]
Other than that, ready for review.

**Comment by jaeilepp** [comment]
I removed the channel name labels if n_channels > 100 as suggested in https://github.com/mne-tools/mne-python/issues/2213. Turns out this makes it quite a bit faster.

**Comment by mainakjas** [comment]
On Tue, Jun 23, 2015 at 11:24 AM, jaeilepp <notifications@github.com> wrote:

> also can the channel names of the bads also be made light gray?
>
> They can be made light gray, but that would mean extra iterations over the
> labels and colors every time view is changed, because of the way the colors
> are defined at the moment. Not worth the effort/decrease in performance in
> my opinion.
>
worth the effort definitely I think. I had this issue myself when I was
working on my own data. Not sure why it would decrease the performance --
maybe we can take a judgement call based on qualitative experience if you
can make a commit (which can be rolled back easily if we don't like it).
Also, why is it so complicated just to change the color of the text ? Does
the code need simplification somewhere?

File: doc/source/whats_new.rst [hunk 3]
```diff
@@ -16,6 +19,7 @@ BUG
 
     - Fix ``mne.stats.bonferroni_correction`` reject mask output to use corrected p-values by `Denis Engemann`_
 
+
```
Comment (mainakjas) [inline doc/source/whats_new.rst]: please remove this extra line

File: doc/source/whats_new.rst [hunk 4]
```diff
@@ -118,8 +122,6 @@ Changelog
 
    - Add support for custom sensor positions when creating Layout objects by `Chris Holdgraf`_
 
-   - Add epochs browser to interactively view and manipulate epochs by `Jaakko Leppakangas`_
```
Comment (mainakjas) [inline doc/source/whats_new.rst]: why?

File: mne/viz/tests/test_epochs.py [hunk 5]
```diff
@@ -104,9 +104,10 @@ def test_plot_epochs():
...
     plt.close('all')
-    epochs[0].plot(picks=[0, 2, 3], scalings=None, trellis=False)
+    fig = epochs[0].plot(picks=[0, 2, 3], scalings=None, trellis=False)
+    fig.canvas.key_press_event('escape')
```
Comment (mainakjas) [inline mne/viz/tests/test_epochs.py]: just noticed. You don't have any tests for interactive dropping of epochs?

File: mne/viz/epochs.py [hunk 6]
```diff
@@ -871,8 +901,52 @@ def _plot_traces(params):
                 False)
     params['ax2'].set_xlim(params['times'][0],
                            params['times'][0] + params['duration'], False)
-    ax.set_yticklabels(tick_list)
-    params['vsel_patch'].set_y(params['ch_start'])
+    if butterfly:
```
Comment (mainakjas) [inline mne/viz/epochs.py]: `if butterfly is True`

File: mne/viz/epochs.py [hunk 7]
```diff
@@ -825,11 +835,18 @@ def plot_epochs_psd(epochs, fmin=0, fmax=np.inf, proj=False, n_fft=256,
 
 def _plot_traces(params):
     """ Helper for plotting concatenated epochs """
+    params['text'].set_visible(False)
     ax = params['ax']
-    n_channels = params['n_channels']
+    butterfly = params['butterfly']
+    if butterfly:
```
Comment (mainakjas) [inline mne/viz/epochs.py]: `if butterfly is True`

File: mne/viz/epochs.py [hunk 8]
```diff
@@ -843,13 +860,27 @@ def _plot_traces(params):
...
-        ch_idx = line_idx + params['ch_start']
+        ch_idx = line_idx + ch_start
         if line_idx >= len(lines):
             break
-        elif ch_idx < len(params['picks']):
-            ch_name = params['ch_names'][ch_idx]
-            tick_list += [ch_name]
-            offset = offsets[line_idx]
+        elif ch_idx < len(params['ch_names']):
+            if butterfly:
```
Comment (mainakjas) [inline mne/viz/epochs.py]: same here

**Comment by mainakjas** [comment]
Bravo, the plot now looks good even if some channel types are missing!

**Comment by mainakjas** [comment]
about color coding, I'd say bad channels -> light gray like in `mne_browse_raw` and bad epochs -> light red like we already had. Where they intersect, just keep light red.

How can I select the bad channels? I don't see it in the docs

File: mne/viz/epochs.py (hunk 7 above)
Comment (jaeilepp) [inline mne/viz/epochs.py]: That's actually against what PEP8 says. Is there a particular reason why 'is True' should be added?

**Comment by jaeilepp** [comment]
You can't select bad channels interactively. They are just shown in different colors if they exist.

File: mne/viz/tests/test_epochs.py (hunk 5 above)
Comment (jaeilepp) [inline mne/viz/tests/test_epochs.py]: Yes there is a test. On line 142 the number of epochs should be one less after the click on axis.

**Comment by mainakjas** [comment]
hmm ... and what was the reason for not allowing bad channel selection interactively?

File: mne/viz/tests/test_epochs.py (hunk 5 above)
Comment (mainakjas) [inline mne/viz/tests/test_epochs.py]: ok cool

**Comment by jaeilepp** [comment]
We are running out of buttons. Left click is for bad epochs and right click for vertical lines.

**Comment by mainakjas** [comment]
Can't we use left click on channel name?

**Comment by jaeilepp** [comment]
Yeah, that's possible.

**Comment by mainakjas** [comment]
ok let's do it then :)

File: doc/source/whats_new.rst (hunk 4 above)
Comment (jaeilepp) [inline doc/source/whats_new.rst]: It was moved to version 10. It was at a wrong place.

Comment (mainakjas) [inline doc/source/whats_new.rst]: oh dear ...

Comment (agramfort) [inline doc/source/whats_new.rst]: are we good here?

**Comment by jaeilepp** [comment]
Ready for review.

**Comment by teonlamont** [comment]
clicking to select bad channels is great! we will need that feature in `raw.plot` but that can be a separate pr.
the butterfly plot is slow for me :( what are people's opinion of just plot the max and min of the good channels with fill and leave the bads as it is.

**Comment by teonlamont** [comment]
just noticed this: dropping bads in the epochs viewer should update the `info['bads']`. this is the behavior we have in `raw.plot`.

**Comment by teonlamont** [comment]
sorry, my last comment was mistaken. the channel amplitudes changed from butterfly to channel view which made the view look different. I think we would want to keep the scaling of the two plots independent.

**Comment by agramfort** [comment]
+1 for merge when Teon is happy

**Comment by teonlamont** [comment]
@agramfort what are your thoughts on the butterfly plot just being the envelope of the signal. it might be time for me to upgrade my hardware but the browsing very sluggish b/c I think it's plotting all the lines in this view but that doesn't really give more information.

**Comment by agramfort** [comment]
I agree that it's dense but it's a quick way to see if one channels goes
wrong compared to others.

**Comment by dengemann** [comment]
@teonlamont I'd rather suggest to collect and implement a low performance config profile that can be selected optionally, e.g. for SSH or old computers.

**Comment by dengemann** [comment]
```Python
with mne.viz.low_performance(**kwarks):
    epochs.plot()
```

**Comment by teonlamont** [comment]
ok cool. i will look into doing so. will be switching back to my desktop soon so this should become less of a problem.
only concern now is independent scaling for the plot. i'm actually ambivalent now so everyone likes it as is then i'm +1 then

**Comment by teonlamont** [comment]
oh, maybe a command line message to remind/notify the user of the drop channel similar to the drop epochs message. would be very useful.

**Comment by dengemann** [comment]
> only concern now is independent scaling for the plot.

What do you mean?

**Comment by jaeilepp** [comment]
Just fixed it.

**Comment by teonlamont** [comment]
> What do you mean?

scaling butterfly plot doesn't affect channel view and vice versa

**Comment by teonlamont** [comment]
for plotting the bad channels, can you make it so that it is in the background for all the channels. in the channel view, it is currently overlapping the previous channel.

**Comment by agramfort** [comment]
Let's merge then

**Comment by jaeilepp** [comment]
Yeah, ready from my part.

**Comment by dengemann** [comment]
Thanks @jaeilepp !!!

**Pull request merged**